woocommerce/woocommerce-ios
# Xcode 15: Fix runtime warning for uploading media files

Closes: #11354 


## What

This PR adds a fix to get rid of the runtime warning for uploading media files on Xcode 15.

## Why
We were sending a non-empty `httpBody` for the upload requests. The compiler complains as it only processes the uploaded data, and the data sent in the `httpBody` is ignored.

## How
The fix is not to encode anything to send to the body of the upload requests. Specifically, I made the `encoding` optional for `DotcomRequest` to skip encoding request body for upload requests.

## Testing instructions

Prerequisite: to confirm the fix, please build and run the app on Xcode 15.
- Log in to a store and navigate to the Products tab.
- Select any existing product or create a new one.
- Upload at least one photo from the device and save the product.
- The photo should be uploaded successfully to the store media library and the product should be saved successfully. No runtime warning should be displayed on Xcode.

## Screenshots

N/A

---
